### DISCUSSION TITLE
Announcement for the dev docs and unifying the sections of the docs.

### DISCUSSION THREAD

**Discussion by Revathyvenugopal162** [ansys/pyansys-dev-guide#217] [General]
In favor of the ongoing modification on multi-doc in the documentation of our projects, Following are the modification proposed for the docs.
1. Add a developer announcement for the documentation in the same way as depreciation warning for older version. Example https://mapdl.docs.pyansys.com/release/0.62/
2. Uniform sections for all documentation. (Getting started, User Guide, API etc..). Keeping the contribution in only one repo(probably in dev.docs.pyansys.com) and link to that by an icon. (like dark/light mode button on the right corner of the docs)
3. Add an icon in documentation to download the pdf version. 
Pinging here @pyansys/pyansys-core and @pyansys/documentation for further discussion.

**Comment by PipKat** [comment]
@Revathyvenugopal162 ,
Would you please clarify this statement in point 2:

Keeping the contribution in only one repo(probably in dev.docs.pyansys.com) and link to that by an icon.

Thanks!

**Comment by Revathyvenugopal162** [comment]
![blue button](https://user-images.githubusercontent.com/104772255/207001201-8a7bd24b-4abd-49ab-82f7-845b378a3acc.PNG)
Yeah sure. Instead of having a section called contribution, we can keep an icon for that, as seen in the screenshot attached. And for most of the cases we can keep that icon linking to a common page. (It will take us the contribution page in pyansys dev guide.)

**Reply by germa89** [reply]
The pyansys dev guide is quite general, whereas some projects might have some specific information about how to contribute. I think that button (in my opinion, very small) should point to a section/chapter in the documentation. In that text, we can then reference the pyansys dev doc.

**Reply by PipKat** [reply]
I agree with German. Several libraries have a Contribute topic with information that is unique to their respective repos/setup. The Contribute topic then links to the general Contribute topic in the PyAnsys Developer's Guide for general guidelines. I also have no include what the icon is supposed to be! It looks like a gavel! It doesn't seem intuitive at all.

**Reply by germa89** [reply]
![image](https://user-images.githubusercontent.com/28149841/207115100-ce43fbce-9a1f-4ae7-a491-36ba9e2deb88.png)

**Comment by germa89** [comment]
Regarding:
> Uniform sections for all documentation.

Which sections are going to be?

I presume it will be:

* Getting started
* User guide
* MAPDL commands
* Examples

Then where we should put the following?

* Learning PyMAPDL
* Troubleshooting
* Contributing and API

I would like these sections to have enough visibility. 

Pinging @Revathyvenugopal162 @jorgepiloto @PipKat for visibility.

**Reply by PipKat** [reply]
First, I do not like "Contributing and API" grouped together in a section. It doesn't make sense to me! All other guides have an "API reference" section. I think "Learning PyMAPDL," which is only one page, should be moved to the "Getting started" section, and "Troubleshooting,. which is two pages, should also be moved to the "Getting started" section. I think we can easily redo the MAPDL "Getting started" section like the [PyFluent "Getting started" section](https://dev.fluent.docs.pyansys.com/getting_started/index.html).

I'd propose that the MAPDL "Getting started" section have these pages: Installation, Versions and interfaces, MAPDL in Docker, PyMAPDL in Julia, PyMAPDL on a Windows Subsytem for Linux and Docker, Troubleshooting, and Frequently asked questions. Given that all but "Issues when importing and exporting numpy arrays" is installation related, "Troubleshooting" could even become a subtopic of the "Installation" page, and then "Issues when importing and exporting numpy arrays" could be incorporated as an entry on the "Frequently asked questions" page.

If you really want more major sections than can fit in the title bar, we need to figure out how to make a two-row title bar because I think a dropdown for sections not visible would lead to these sections being overlooked. My suggestions would lead us to these sections, with only "MAPDL commands" being non-standard:

- Getting started
- User guide
- MAPDL commands
- API reference
- Examples
- Contribute

**Reply by germa89** [reply]
I agree on everything with @PipKat  except on:

>  Given that all but "Issues when importing and exporting numpy arrays" is installation related, "Troubleshooting" could even become a subtopic of the "Installation" page, and then "Issues when importing and exporting numpy arrays" could be incorporated as an entry on the "Frequently asked questions" page.

I think ``Troubleshooting`` should be very visible, and it is my intention to fit there all the common-general issues to launch/use PyMAPDL. Hence I would propose something like this:

- Getting started
  - {{ Previous Sections }}
  - Learning
  - Troubleshooting
  - Frequency Asked Questions
- User guide
- MAPDL commands
- API reference
- Examples
- Contribute

Furthermore, if we want to (I dont), we could move `MAPDL commands` to `User guide`. But because (I think) they are very used, I wouldn't move it.

**Reply by PipKat** [reply]
@germa89 I'm perfectly happy with your latest proposal. I think the MAPDL commands should be at the highest level as well (and am fine with a separate "Troubleshooting" section if it is to grow as planned. (In this case, it might be better located in the "User guide" section rather than the "Getting started" section though. (We could reference it in an FAQ asking "Where do you find information on resolving common issues?" or something like that.

Also, based on yesterday's Pronovix conversation, we might have to look at how the MAPDL commands are being imported for inclusion in the PyMAPDL documentation to ensure that this command section will be a part of the JSON file created for porting to the developer portal. I know Alex is trying to convince the MAPDL doc team to allow the source files to be placed in a GitHub repo so that a web editor ([Oxygen XML Web Author](https://www.oxygenxml.com/xml_web_author.html)) can be used to "inner source" changes to this content. If this proposal is approved, I believe that the MAPDL commands content would then be part of the Sphinx documentation generation, which would solve the problem.

**Reply by germa89** [reply]
>  (In this case, it might be better located in the "User guide" section rather than the "Getting started" section though. (We could reference it in an FAQ asking "Where do you find information on resolving common issues?" or something like that.

I like the idea of referencing troubleshooting in the FAQ section. 

I think this should be detailed in the dev guide, and we should also be consistent with where this sections are (FAQ, and Troubleshoot included). For PyMAPDL I feel they belong to ``Getting started``, but I will go with the general common sense we can get from this discussion.

So, where ``Troubleshoot`` and ``FAQ`` should be? In ``Getting started`` or ``User guide`` ??

**Reply by germa89** [reply]
Additionally, we should include this structure in the ansys-template package.

**Reply by germa89** [reply]
Interesting case is also ``Contribution``, @akaszynski  mentioned he would like it to be as an icon. I also think it should be searchable through the previous defined structure.

I believe the API is more about developers, so it is about contributing. That is why I merged API with Contributing in PyMAPDL. But this seems a bit confusing for some people.
Then I would propose to link Contribution in ``Getting Started``, and also at the top right icon (we should change the icon though).

**Reply by PipKat** [reply]
@germa89 Tom Johnson is a highly respected blogger on technical writing. In the following post, he makes some excellent points about where troubleshooting sections (which really provide answers to FAQs) should go: 
https://idratherbewriting.com/2010/07/08/separating-basic-from-advanced-topics-how-twitter-organizes-their-help-organizing-content-18/

I think it would be cool if we create some type of directive with a cool icon for a Troubleshooting subsection and then put it within the relevant topic, as suggested by Kathy Sierra in this post from several years ago: [The best user manuals EVER](http://headrush.typepad.com/creating_passionate_users/2007/03/the_best_user_t.html).

**Reply by germa89** [reply]
I'm going (for the moment) with this:

![image](https://user-images.githubusercontent.com/28149841/207425711-5d3db355-38e8-4a73-87e4-dd27ca2fff4b.png)

As detailed in https://github.com/pyansys/pymapdl/pull/1689

Tom Johnson is proposing to split troubleshooting across the documentation, so it is easy to find it. I do not like much that. I understand the point though.
I rather have links inside the documentation to the troubleshoot section.

**Comment by germa89** [comment]
I'm going to detail my proposal for documentation structure:

- Getting started
  - Learning
  - Frequency Asked Questions
  - Contribute
- User guide
  - Troubleshooting
- MAPDL commands
- API reference
- Examples

**Reply by germa89** [reply]
I'm pushing this scheme in PyMAPDL. We can adjust it later.

https://github.com/pyansys/pymapdl/pull/1689

**Reply by germa89** [reply]
For your info, the following projects have the next structure: 

## Numpy
Link: https://numpy.org/doc/1.23/
* User Guide
* API reference
* Development
* Release notes
* Learn

## Pandas
Link: https://pandas.pydata.org/docs/
* Getting started
* User Guide
* API reference
* Development
* Release notes

## Pyvista
Link: https://docs.pyvista.org/
* Getting Started
* User Guide
* Examples
* API Reference
* Extras


Surprisingly all of them have a ``Development`` tab which talk about contributing, guidelines, benchmarking, etc 
Numpy dev page is https://numpy.org/doc/1.23/dev/index.html

Maybe we should consider to something like this too. The icon might be a bit to hidden. 

I dont like the release notes on the main though.

**Comment by germa89** [comment]
Furthermore... I believe we can change the link of the contribution button using one of these:

https://fontawesome.com/v4/icons/

I have selected the following:

* https://fontawesome.com/v4/icon/wrench [My preferred]
* https://fontawesome.com/v4/icon/pencil
* https://fontawesome.com/v4/icon/code
* https://fontawesome.com/v4/icon/terminal
* https://fontawesome.com/v4/icon/star
* https://fontawesome.com/v4/icon/heart
* https://fontawesome.com/v4/icon/files-o
* https://fontawesome.com/v4/icon/cogs
* https://fontawesome.com/v4/icon/cog

**Reply by germa89** [reply]
Also, because of how the icons are set in the ``conf.py``
```py
....
{
            "name": "Contribute",
            "url": "https://mapdl.docs.pyansys.com/release/dev/getting_started/contribution.html",
            "icon": "fa fa-wrench",
        },
...
```
We need to use the explicit hard link... Hence I would propose to use the dev version so we don't need to change it accordingly to the version.

**Reply by germa89** [reply]
Pinging @Revathyvenugopal162 @jorgepiloto for feedback on this.

**Reply by germa89** [reply]
I think we could hack it to point to the Contribution in latest stable... but I dont think it is a big deal.
